Sage/carbon
# fix(box): ensure z-index is set when sticky or fixed positioned

fix #5572

### Proposed behaviour


Ensures that the any `Box` container that has `sticky` or `fixed` positioning has a z-index set to prevent it being overlaid by other elements that are relatively positioned

### Current behaviour


Any `Box` container that has `sticky` or `fixed` positioning has no z-index set

### Checklist




#### QA


### Additional context



### Testing instructions


See sandbox in linked issue below

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
